bamboo/detect-pr.js

```diff
@@ -0,0 +1,52 @@
+/* eslint-disable node/no-unpublished-require */
+/* eslint-disable no-console */
+
+'use strict';
+
+const graphql = require('@octokit/graphql');
```

Comment:
This query uses github's v4 API which is moving to GraphQL.    Using same approach as the stack lock tool to pass values back to the shell scripts. 

This query grabs the current ref, takes the most recent commit log, pulls any pull requests associated with it and if there are ,returns success (with a log output on which PRs were detected).   

It's worth noting that this *will* trigger if for some reason a commit SHA that is the HEAD of an open PR is somehow attached to the head of a branch and triggers a build.   This seems like an unusual scenario, but mentioning it here for reviewers to consider.      I'll be adding that to a wiki/demo page for the team in the future as well.